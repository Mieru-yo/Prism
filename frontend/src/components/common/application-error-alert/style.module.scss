@use "sass:color";
/*!
 * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
 *
 * SPDX-License-Identifier: AGPL-3.0-only
 */

.alert {
  $color: #5A0F04;
  display: flex;
  --bs-alert-color: #{color.adjust($color, $lightness: 80%)};
  --bs-alert-bg: #{$color};
  --bs-alert-border-color: #{color.adjust($color, $lightness: 20%)};
  flex-direction: row;
}

.logo {
  flex: 0 0 50px;
  height: 48px;
  margin-right: 8px;
}
