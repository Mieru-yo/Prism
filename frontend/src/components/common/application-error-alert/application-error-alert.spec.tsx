/*
 * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
 *
 * SPDX-License-Identifier: AGPL-3.0-only
 */

import { render } from '@testing-library/react'
import { ApplicationErrorAlert } from './application-error-alert'
import type { AlertIconProps } from './alert-icon'
import { describe, expect, it, vitest, beforeAll, afterAll } from 'vitest'
import { vi } from 'vitest'

vi.mock('./alert-icon', () => ({
  AlertIcon: (props: AlertIconProps) => `This is a mock for "AlertIcon". Props: ${JSON.stringify(props)}`
}))

describe('ApplicationErrorAlert', () => {
  it('renders correctly', () => {
    const view = render(
      <ApplicationErrorAlert className={'test-class'}>
        <span>Test Child</span>
      </ApplicationErrorAlert>
    )

    expect(view.container).toMatchSnapshot()
  })
})
