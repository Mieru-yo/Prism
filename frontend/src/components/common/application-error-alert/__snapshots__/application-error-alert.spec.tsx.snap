// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ApplicationErrorAlert > renders correctly 1`] = `
<div>
  <div
    class="fade _alert_663ce3 test-class alert alert-primary show"
    role="alert"
  >
    <p
      class="d-flex align-items-center"
    >
      This is a mock for "AlertIcon". Props: {"className":"_logo_663ce3"}
      <span>
        <span>
          Test Child
        </span>
      </span>
    </p>
  </div>
</div>
`;
